lib/reverse_adoc/converters/figure.rb

```diff
@@ -1,12 +1,14 @@
 module ReverseAdoc
   module Converters
     class Figure < Base
-      def convert(node, state = {})
+      def to_coradoc(node, state = {})
         id = node['id']
-        anchor = id ? "[[#{id}]]\n" : ""
         title = extract_title(node)
-        title = ".#{title}\n" unless title.empty?
-        "\n\n#{anchor}#{title}====\n" << treat_children(node, state).strip << "\n====\n\n"
+        content = treat_children(node, state).strip
+        Coradoc::Document::Block.new(title, lines: content.lines, type: :example, id: id)
```

Comment:
The `content.lines` should be converted to Coradoc objects before putting in.

e.g.
```html
<figure>
  <img src="pic_trulli.jpg" alt="Trulli" style="width:100%">
  <figcaption>Fig.1 - Trulli, Puglia, Italy.</figcaption>
</figure>
```